tianosouza/to_converter
# Feat/converter to

### Add Unit Toggle Functionality and Measurement Conversion Adjustments

Description
This pull request implements the functionality to toggle between px and rem units in the measurement conversion form. Additionally, adjustments were made to the Stimulus controller and tests to ensure that the conversion logic works correctly and that the input and output units are updated as expected.

Main Changes:
New Stimulus Controller (toggle-units_controller.js):

Implemented logic to toggle units between px and rem and vice versa.
Updated the values of the input and output fields when clicking the "Toggle Units" button.
Added an updateLabels function to correctly reflect the units in the form labels.
Adjustments to Rails Controller (ConverterToController):

Modified the convert method to ensure that the correct unit is displayed in the conversion result.
Displayed the unit of measurement along with the converted value in the view.
Updates to the View:

Added a "Toggle Units" button that triggers the unit toggle logic via Stimulus.
Applied basic styling to the interface using Tailwind CSS.
Test Adjustments:

Refactored RSpec tests to validate the conversion logic from px to rem and from rem to px.
Added tests to ensure the unit toggle logic functions correctly.
![image](https://github.com/user-attachments/assets/0d45e465-9876-4865-a528-027b0d99b864)

How to Test:
Access the application and navigate to the measurement conversion form.
Enter a value and select the desired units.
Click "Toggle Units" and verify that the input and output units are correctly swapped.
Submit the form and check if the converted value and its corresponding unit are displayed correctly.

Form in view:
![image](https://github.com/user-attachments/assets/1f67b66f-52b1-4d89-9a7d-f5a2924f8940)
